### ISSUE TITLE
Support coqffi in coq.theory stanza

### ISSUE THREAD

**Issue by lthms** [ocaml/dune#3995]
Hi,

Lately, I have been working on  [`coqffi`](https://github.com/coq-community/coqffi), a tool to generate “FFI modules” to OCaml functions. It is my hope that `coqffi` will eventually be adopted by the Coq community, as a convenient way to mix Coq and OCaml code.

`coqffi` takes a `cmi` file as its input, and generate a Coq module. I have been using it with `dune` with a lot of success, as demonstrated by the `examples` directory of the repo (see the [dune file](https://github.com/coq-community/coqffi/blob/main/examples/src/dune)).

For instance,

```dune
(rule
  (target File.v)
  (action (run coqffi -finterface %{cmi:file} -o %{target})))
```

The fact that `dune` understands out of the box that `File.v` is a dependency of a theory declared with the `coq.theory` stanza is truly amazing. Thanks!

I was wondering if, when the time is right, it would be possible to add a dedicated field in `dune` to the `coq.theory` for instance, for `coqffi` to be more easily accessible to Coq users.

## Desired Behavior

Basically, the idea is to have a more straightforward way to generate FFI modules, for instance, `ffi_modules`.

## Example

```dune
(library MyLib)

(coq.theory
  (theory Test)
  (ffi_modules (MyLib.Test))
```

Which would means (1) generate the `Test.v` module from the `cmi` of the `Test` module from the `MyLib` library, and (2) add this module to the list of modules of the coq theory `Test`.

We could imagine being able to provide `coqffi` arguments, like

```dune
(coq.theory
  (theory Test)
  (ffi_modules ((MyLib.Test :flags -finterface))))
```

_________

So to wrap-up, do you think such a thing is possible? Is there a way for me to make it happen (which is not hacking `dune` to add the feature, but more from the `coqffi` side. The first obvious step is to publish a stable version of `coqffi` of course, but in addition to that, is there anything that comes to your minds?

Thanks for your feedback!

**Comment by ejgallego** [comment]
Hi @lthms , and thanks for your proposal. I think indeed this should be OK to do and not too hard at all.

I'll be happy to hack on it together if you want at the ongoing CUDW.

**Comment by lthms** [comment]
I would love to! I guess CUDW is the Coq User and Developer Workshop?

**Comment by ejgallego** [comment]
Yes sorry! We have a stream on Zulip so we can coordinate there.

**Comment by Alizter** [comment]
I have a branch supporting coqffi. I haven't been able to test it yet since I am not sure how to set up coqffi as a test suite dep. It is currently sitting on top of my PR #6190 for coq-of-ocaml support, but in theory it could be rebased if there is interest.

**Comment by lthms** [comment]
Your PR only mention `coq-of-ocaml`, are you sure you don’t mix the two? (I’d love to see someone else adding support for coqffi in `dune`, for sure ahah)

**Comment by Alizter** [comment]
@lthms I also have coqffi but I haven't been able to push yet. Let me rebase and show it.

**Comment by Alizter** [comment]
Here it is: https://github.com/ocaml/dune/pull/6248

**Comment by ejgallego** [comment]
coqffi doesn't seem active anymore, feel free to re-open if that's not the case.
